crates/ty_ide/src/goto_definition.rs

```diff
@@ -345,6 +345,46 @@ class MyClass: ...
         assert_snapshot!(test.goto_definition(), @"No goto target found");
     }
 
+    #[test]
+    fn goto_definition_stub_map_both_stubbed() {
+        // If the .pyi and the .py both define a stub, prefer the .py version
```

Comment:
I find this test (and the wording describing it here) a bit confusing. In an informal sense, I guess you could call `class MyClass: ...` in a `.py` file a "stub" (in that the class has no body, and it uses an ellipsis, like stubs do), but it's not really a "stub" in any sense described by the typing spec, so I wouldn't call it one here. And I'm not sure what value this test adds, because I don't think we would ever treat `class MyClass: ...` any differently from a more complex class definition, for the purposes of goto-definition?

Reply:
I think my comment is kind of driving in the opposite direction here? That is, I think that `class MyClass: ...` in a `.py` file is not in any meaningful sense a "stub". It is a fully valid implementation/definition of `MyClass` (just as much as `class MyClass:` followed by a three-hundred-line body), and we shouldn't treat it differently in any way. Goto-definition should happily go to it. So I don't really even understand what is noteworthy or test-worthy about this case.

Reply:
I'm gonna leave it in the PR for now so others have a chance to chime in here (I'm fine with whatever).

Reply:
That's fine, but looking into Alex's comment where I think this test came from, I'm pretty sure the fact that both definitions use `class Foo: ...` in that example is entirely incidental, not part of Alex's point. The point of that comment was simply that we were wrongly failing to prefer `module.py` over `module.pyi`. You already fixed that (by pushing `mode` deeper), and plenty of other tests here validate that fix already. So I'm pretty confident this test can be removed or reworked per my comment above, without contradicting anything Alex was suggesting.